src/app/core/service/flight.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable, OnInit } from '@angular/core';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { BehaviorSubject, Observable, catchError, map, tap, throwError } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { Flight } from '../interface/flight';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FlightService {
+  private flightsSubject: BehaviorSubject<Flight[]> = new BehaviorSubject<
+    Flight[]
+  >([]);
+  public flights$: Observable<Flight[]> = this.flightsSubject.asObservable();
+
+  constructor(private http: HttpClient) {}
+
+  getAllFlights() {
+    return this.http.get(`${environment.apiUrl}/flights`).pipe(
+      catchError(this.errorHandler),
+      map((response) => {
+        let flights: Flight[] = [];
+        for (let key in response) {
+          if (response.hasOwnProperty(key)) {
+            flights.push((response as Record<string, Flight>)[key]);
+          }
+        }
+        return flights;
+      }),
+      tap((data) => {
+        this.flightsSubject.next(data);
+      })
+    );
+  }
+
+  updateFlight(requestBody: any, flightId: any) {
+    return this.http
+      .put(`${environment.apiUrl}/flights/${flightId}`, requestBody)
+      .pipe(
+        catchError(this.errorHandler),
+        tap((data) => {
+          let flights: Flight[] = this.flightsSubject.getValue();
+
+          flights = flights.map((flight: Flight) => {
```

Comment:
use filter function

Reply:
@aky-wtag vai i have removed this tap method